examples/petsc/bps.c

```diff
@@ -497,17 +601,22 @@ int main(int argc, char **argv) {
   // Gather RHS
   ierr = VecRestoreArray(rhsloc, &r); CHKERRQ(ierr);
   ierr = VecZeroEntries(rhs); CHKERRQ(ierr);
-  ierr = VecScatterBegin(ltog0, rhsloc, rhs, ADD_VALUES, SCATTER_FORWARD);
+  ierr = VecScatterBegin(ltog, rhsloc, rhs, ADD_VALUES, SCATTER_FORWARD);
   CHKERRQ(ierr);
-  ierr = VecScatterEnd(ltog0, rhsloc, rhs, ADD_VALUES, SCATTER_FORWARD);
+  ierr = VecScatterEnd(ltog, rhsloc, rhs, ADD_VALUES, SCATTER_FORWARD);
   CHKERRQ(ierr);
   CeedVectorDestroy(&rhsceed);
 
   ierr = KSPCreate(comm, &ksp); CHKERRQ(ierr);
   {
     PC pc;
     ierr = KSPGetPC(ksp, &pc); CHKERRQ(ierr);
-    ierr = PCSetType(pc, PCNONE); CHKERRQ(ierr);
+    if (bpChoice == CEED_BP1 || bpChoice == CEED_BP2) {
+      ierr = PCSetType(pc, PCJACOBI); CHKERRQ(ierr);
+      ierr = PCJacobiSetType(pc, PC_JACOBI_ROWSUM); CHKERRQ(ierr);
```

Comment:
This could become a run-time option. Nothing in [the spec](https://ceed.exascaleproject.org/bps/) says that Jacobi preconditioning should be used (though it speeds up convergence for these cases.